src/compiler/validation.jl

```diff
@@ -48,9 +48,20 @@ end
 
 function check_ir!(job, errors, imported, f::LLVM.Function, known_fns)
     calls = []
+    isInline = API.EnzymeGetCLBool(cglobal((:EnzymeInline, API.libEnzyme))) != 0
     for bb in blocks(f), inst in instructions(bb)
         if isa(inst, LLVM.CallInst)
             push!(calls, inst)
+        # remove illegal invariant.load and jtbaa_const invariants
+        elseif isInline && isa(inst, LLVM.LoadInst)
```

Comment:
we should probably land this separately from the more experimental finalizer change?